docker/docker_image_dest.go

```diff
@@ -761,7 +783,17 @@ func (d *dockerImageDestination) putSignaturesToSigstoreAttachments(ctx context.
 			return err
 		}
 		sigDesc.Annotations = annotations
-		ociManifest.Layers = append(ociManifest.Layers, sigDesc)
+		switch v := genManifest.(type) {
+		case *manifest.OCI1:
+			v.Layers = append(v.Layers, sigDesc)
+		case *manifest.Schema2:
+			v.LayersDescriptors = append(v.LayersDescriptors, manifest.Schema2Descriptor{
```

Comment:
This discards annotations (schema2 manifests don’t have annotations at all), and as a result, the signature is unusable.